packages/server/lib/hooks/hooks.ts

```diff
@@ -219,37 +226,41 @@ export const connectionRefreshFailed = async ({
     logCtx: LogContext;
     action: 'token_refresh' | 'connection_test';
 }): Promise<void> => {
-    await errorNotificationService.auth.create({
-        type: 'auth',
-        action,
-        connection_id: connection.id,
-        log_id: logCtx.id,
-        active: true
-    });
+    try {
+        await errorNotificationService.auth.create({
+            type: 'auth',
+            action,
+            connection_id: connection.id,
+            log_id: logCtx.id,
+            active: true
+        });
 
-    const webhookSettings = await externalWebhookService.get(environment.id);
+        const webhookSettings = await externalWebhookService.get(environment.id);
 
-    void sendAuthWebhook({
-        connection,
-        environment,
-        webhookSettings,
-        auth_mode: provider.auth_mode,
-        operation: 'refresh',
-        error: authError,
-        success: false,
-        providerConfig: config,
-        account
-    });
+        void sendAuthWebhook({
+            connection,
+            environment,
+            webhookSettings,
+            auth_mode: provider.auth_mode,
+            operation: 'refresh',
+            error: authError,
+            success: false,
+            providerConfig: config,
+            account
+        });
 
-    await slackService.reportFailure({
-        account,
-        environment,
-        connection,
-        name: connection.connection_id,
-        type: 'auth',
-        originalActivityLogId: logCtx.id,
-        provider: config.provider
-    });
+        await slackService.reportFailure({
+            account,
+            environment,
+            connection,
+            name: connection.connection_id,
+            type: 'auth',
+            originalActivityLogId: logCtx.id,
+            provider: config.provider
+        });
+    } catch (err) {
+        report(new Error('refresh_failed_hook_failed', { cause: err }), { id: connection.id });
```

Comment:
same question about the single try catch

Reply:
also changed